From 8bb708ada3a83af05ea6b566f5f27431721dd272 Mon Sep 17 00:00:00 2001
From: Samuel Dionne-Riel <samuel@dionne-riel.com>
Date: Fri, 18 Jun 2021 21:13:00 -0400
Subject: [PATCH] [HACK] touchscreen: Skip loading firmware

This saves 60 seconds waiting for the touchscreen to work. We don't have
the firmware files available *yet*.

When we'll have the firmware files available for the phone, the next
step will be to make this firmware file available to stage-1; either by
copying it into it (redistributable?) or by mounting a vendor partition.
---
 drivers/input/touchscreen/stm/fts_lib/ftsFlash.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/input/touchscreen/stm/fts_lib/ftsFlash.c b/drivers/input/touchscreen/stm/fts_lib/ftsFlash.c
index 724588955dd3c..9edc27cd5e8a6 100644
--- a/drivers/input/touchscreen/stm/fts_lib/ftsFlash.c
+++ b/drivers/input/touchscreen/stm/fts_lib/ftsFlash.c
@@ -67,6 +67,9 @@ extern SysInfo systemInfo;	/* /< forward declaration of the global variable
   */
 int getFWdata(const char *pathToFile, u8 **data, int *size)
 {
+	pr_info("[HACK] Skipping getFWdata entirely...\n");
+	return ERROR_FILE_NOT_FOUND;
+
 	const struct firmware *fw = NULL;
 	struct device *dev = getDev();
 	struct fts_ts_info *info = NULL;
-- 
2.29.2

